node/src/Pos/DB/Block.hs

```diff
@@ -287,8 +287,9 @@ type MonadBlockDBWrite ssc m
 -- Pure implementation
 ----------------------------------------------------------------------------
 
+-- [CSL-1493] Should we make this function return an `Either e (Block ssc, Undo)`?
```

Comment:
In this particular case `unsafeDeserialize'` is not so critical, because we are decoding data from our DB which we control by ourselves. But I would like to avoid it anyway. We have `DBMalformed` exception for the cases when our DB is malformed (so we don't rely that it's always valid and handle it gracefully).
`decodeOrFailPureDB` is used only in `dbGetBlundPure` which has (indirectly `MonadThrow` constraint), so it can be thrown from there (using `throwM`).
tl;dr: yes, I would like to see `Either` here (or `MonadThrow` constraint).